x/market/keeper/deal.go

```diff
@@ -0,0 +1,120 @@
+package keeper
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	gogotypes "github.com/gogo/protobuf/types"
+	"github.com/medibloc/panacea-core/v2/x/market/types"
+)
+
+func (k Keeper) CreateNewDeal(ctx sdk.Context, owner sdk.AccAddress, deal types.Deal) (uint64, error) {
+	dealId := k.GetNextDealNumberAndIncrement(ctx)
+
+	newDeal, err := k.newDeal(dealId, deal)
+	if err != nil {
+		return 0, err
+	}
+
+	var coins sdk.Coins
+	coins = append(coins, *deal.GetBudget())
+
+	acc := k.accountKeeper.GetAccount(ctx, sdk.AccAddress(newDeal.GetDealAddress()))
+	if acc != nil {
+		return 0, sdkerrors.Wrapf(types.ErrDealAlreadyExist, "deal %d already exist", dealId)
+	}
```

Comment:
It seems that this logic is only for checking if the deal module account already exist. Did I understand it correctly?

Then, I think we need to use `AccDealAddressFromBech32(newDeal.GetDeadAddress())` (or something similar).

Reply:
I'm not sure since I'm not familiar with module accounts. Please check if the code above works well as you indended.

Reply:
Yes, This is my mistake. I didn't change this code yet. I'll change it to ```AccDealAddressFromBech32(newDeal.GetDealAddress())```. Thank you for your feedback.